Problems during installation on linux

I met the same problem for several times during installation on linux and have no idea how to solve this when I run `python -m pip install -e .`

```
_ERROR: Command errored out with exit status 1:
     command: /home/lianqiuyu/anaconda3/bin/python -c 'import sys, setuptools, tokenize; sys.argv[0] = '"'"'/tmp/pip-install-wu2k721r/python-javabridge_5d7b2d8af62842c8875ab526c4ce1bee/setup.py'"'"'; __file__='"'"'/tmp/pip-install-wu2k721r/python-javabridge_5d7b2d8af62842c8875ab526c4ce1bee/setup.py'"'"';f=getattr(tokenize, '"'"'open'"'"', open)(__file__);code=f.read().replace('"'"'\r\n'"'"', '"'"'\n'"'"');f.close();exec(compile(code, __file__, '"'"'exec'"'"'))' egg_info --egg-base /tmp/pip-pip-egg-info-ihuvetlh
         cwd: /tmp/pip-install-wu2k721r/python-javabridge_5d7b2d8af62842c8875ab526c4ce1bee/
    Complete output (9 lines):
    Traceback (most recent call last):
      File "<string>", line 1, in <module>
      File "/tmp/pip-install-wu2k721r/python-javabridge_5d7b2d8af62842c8875ab526c4ce1bee/setup.py", line 412, in <module>
        ext_modules=ext_modules(),
      File "/tmp/pip-install-wu2k721r/python-javabridge_5d7b2d8af62842c8875ab526c4ce1bee/setup.py", line 142, in ext_modules
        library_dirs = [os.path.dirname(jvm_so)]
      File "/home/lianqiuyu/anaconda3/lib/python3.7/posixpath.py", line 156, in dirname
        p = os.fspath(p)
    TypeError: expected str, bytes or os.PathLike object, not NoneType
    ----------------------------------------_

```

Hi @manthey ,
I have a related, but not quite the same problem. I'm trying to install from source on a new machine and I am getting this error .. seems to stem from CMAKE compilation .. any idea what might be happening? Thanks!

```
Running setup.py develop for histomicstk
    ERROR: Command errored out with exit status 1:
     command: /usr/bin/python -c 'import io, os, sys, setuptools, tokenize; sys.argv[0] = '"'"'/opt/mtageld/HistomicsTK/setup.py'"'"'; __file__='"'"'/opt/mtageld/HistomicsTK/setup.py'"'"';f = getattr(tokenize, '"'"'open'"'"', open)(__file__) if os.path.exists(__file__) else io.StringIO('"'"'from setuptools import setup; setup()'"'"');code = f.read().replace('"'"'\r\n'"'"', '"'"'\n'"'"');f.close();exec(compile(code, __file__, '"'"'exec'"'"'))' develop --no-deps --user --prefix=
         cwd: /opt/mtageld/HistomicsTK/
    Complete output (106 lines):
    Not searching for unused variables given on the command line.
    CMake Deprecation Warning at CMakeLists.txt:1 (cmake_minimum_required):
      Compatibility with CMake < 2.8.12 will be removed from a future version of
      CMake.
    
      Update the VERSION argument <min> value or use a ...<max> suffix to tell
      CMake that the project does not need compatibility with older versions.
    
    
    -- The C compiler identification is GNU 7.5.0
    -- Detecting C compiler ABI info
    -- Detecting C compiler ABI info - done
    -- Check for working C compiler: /usr/bin/cc - skipped
    -- Detecting C compile features
    -- Detecting C compile features - done
    -- The CXX compiler identification is GNU 7.5.0
    -- Detecting CXX compiler ABI info
    -- Detecting CXX compiler ABI info - done
    -- Check for working CXX compiler: /usr/bin/c++ - skipped
    -- Detecting CXX compile features
    -- Detecting CXX compile features - done
    -- Configuring done
    -- Generating done
    -- Build files have been written to: /opt/mtageld/HistomicsTK/_cmake_test_compile/build
    -- The C compiler identification is GNU 7.5.0
    -- The CXX compiler identification is GNU 7.5.0
    -- Detecting C compiler ABI info
    -- Detecting C compiler ABI info - done
    -- Check for working C compiler: /usr/bin/cc - skipped
    -- Detecting C compile features
    -- Detecting C compile features - done
    -- Detecting CXX compiler ABI info
    -- Detecting CXX compiler ABI info - done
    -- Check for working CXX compiler: /usr/bin/c++ - skipped
    -- Detecting CXX compile features
    -- Detecting CXX compile features - done
    -- Found PythonInterp: /usr/bin/python (found version "3.8.10")
    -- Found PythonLibs: libpython3.8.a
    -- Found Cython: /tmp/pip-build-env-_b8oxhp6/overlay/bin/cython
    -- Found NumPy: /tmp/pip-build-env-_b8oxhp6/overlay/lib/python3.8/site-packages/numpy/core/include (found version "1.20.3")
    -- Performing Test Weak Link MODULE -> SHARED (gnu_ld_ignore) - Success
    _modinit_prefix:PyInit_
    _modinit_prefix:PyInit_
    NUMPY_INCLUDE_DIRS:/tmp/pip-build-env-_b8oxhp6/overlay/lib/python3.8/site-packages/numpy/core/include
    _modinit_prefix:PyInit_
    -- Configuring done
    -- Generating done
    CMake Warning:
      Manually-specified variables were not used by the project:
    
        SKBUILD
    
    
    -- Build files have been written to: /opt/mtageld/HistomicsTK/_skbuild/linux-x86_64-3.8/cmake-build
    [1/10] Generating CXX source histomicstk/features/_compute_marginal_glcm_probs_cython.cxx
    [2/10] Building CXX object histomicstk/features/CMakeFiles/_compute_marginal_glcm_probs_cython.dir/_compute_marginal_glcm_probs_cython.cxx.o
    FAILED: histomicstk/features/CMakeFiles/_compute_marginal_glcm_probs_cython.dir/_compute_marginal_glcm_probs_cython.cxx.o
    /usr/bin/c++ -D_compute_marginal_glcm_probs_cython_EXPORTS -I/usr/include/python3.8 -I/tmp/pip-build-env-_b8oxhp6/overlay/lib/python3.8/site-packages/numpy/core/include -O3 -DNDEBUG -fPIC -MD -MT histomicstk/features/CMakeFiles/_compute_marginal_glcm_probs_cython.dir/_compute_marginal_glcm_probs_cython.cxx.o -MF histomicstk/features/CMakeFiles/_compute_marginal_glcm_probs_cython.dir/_compute_marginal_glcm_probs_cython.cxx.o.d -o histomicstk/features/CMakeFiles/_compute_marginal_glcm_probs_cython.dir/_compute_marginal_glcm_probs_cython.cxx.o -c histomicstk/features/_compute_marginal_glcm_probs_cython.cxx
    histomicstk/features/_compute_marginal_glcm_probs_cython.cxx:6:10: fatal error: Python.h: No such file or directory
     #include "Python.h"
              ^~~~~~~~~~
    compilation terminated.
    [3/10] Generating CXX source histomicstk/segmentation/nuclear/_max_clustering_cython.cxx
    [4/10] Generating CXX source histomicstk/segmentation/label/_trace_object_boundaries_cython.cxx
    ninja: build stopped: subcommand failed.
      File "/tmp/pip-build-env-_b8oxhp6/overlay/lib/python3.8/site-packages/skbuild/setuptools_wrap.py", line 589, in setup
        cmkr.make(make_args, env=env)
      File "/tmp/pip-build-env-_b8oxhp6/overlay/lib/python3.8/site-packages/skbuild/cmaker.py", line 496, in make
        raise SKBuildError(
    An error occurred while building with CMake.
      Command:
        cmake --build . --target install --config Release --
      Source directory:
        /opt/mtageld/HistomicsTK
      Working directory:
        /opt/mtageld/HistomicsTK/_skbuild/linux-x86_64-3.8/cmake-build
    Please see CMake's output for more information.
    
    
    --------------------------------------------------------------------------------
    -- Trying "Ninja" generator
    --------------------------------
    ---------------------------
    ----------------------
    -----------------
    ------------
    -------
    --
    --
    -------
    ------------
    -----------------
    ----------------------
    ---------------------------
    --------------------------------
    -- Trying "Ninja" generator - success
    --------------------------------------------------------------------------------
    
    Configuring Project
      Working directory:
        /opt/mtageld/HistomicsTK/_skbuild/linux-x86_64-3.8/cmake-build
      Command:
        cmake /opt/mtageld/HistomicsTK -G Ninja -DCMAKE_INSTALL_PREFIX:PATH=/opt/mtageld/HistomicsTK/_skbuild/linux-x86_64-3.8/cmake-install -DPYTHON_EXECUTABLE:FILEPATH=/usr/bin/python -DPYTHON_VERSION_STRING:STRING=3.8.10 -DPYTHON_INCLUDE_DIR:PATH=/usr/include/python3.8 -DPYTHON_LIBRARY:FILEPATH=libpython3.8.a -DSKBUILD:BOOL=TRUE -DCMAKE_MODULE_PATH:PATH=/tmp/pip-build-env-_b8oxhp6/overlay/lib/python3.8/site-packages/skbuild/resources/cmake -DCMAKE_BUILD_TYPE:STRING=Release
    
    Traceback (most recent call last):
    
    ----------------------------------------
ERROR: Command errored out with exit status 1: /usr/bin/python -c 'import io, os, sys, setuptools, tokenize; sys.argv[0] = '"'"'/opt/mtageld/HistomicsTK/setup.py'"'"'; __file__='"'"'/opt/mtageld/HistomicsTK/setup.py'"'"';f = getattr(tokenize, '"'"'open'"'"', open)(__file__) if os.path.exists(__file__) else io.StringIO('"'"'from setuptools import setup; setup()'"'"');code = f.read().replace('"'"'\r\n'"'"', '"'"'\n'"'"');f.close();exec(compile(code, __file__, '"'"'exec'"'"'))' develop --no-deps --user --prefix= Check the logs for full command output.
The command '/bin/bash -c git clone https://github.com/DigitalSlideArchive/HistomicsTK/ &&     cd HistomicsTK/ &&     python -m pip install setuptools-scm Cython>=0.25.2 scikit-build>=0.8.1         cmake>=0.6.0 numpy>=1.12.1  &&     python -m pip install -e . --find-links https://girder.github.io/large_image_wheels' returned a non-zero code: 1

```

--- a/.github/workflows/ci.yml
+++ b/.github/workflows/ci.yml
@@ -1,0 +1,119 @@
+name: Build
+
+on:
+  push:
+  # pull_request:
+  #   branches:
+  #     - master
+  release:
+    types:
+      - created
+  # schedule:
+  #   - cron: "0 7 * * 1"
+
+concurrency:
+  group: ${{ github.ref }}
+  cancel-in-progress: ${{ github.ref != 'refs/heads/master' }}
+
+jobs:
+  test:
+    runs-on: ubuntu-latest
+    # Runs can be slow, so don't use resources unless we mean to publish
+    if: github.repository == 'DigitalSlideArchive/HistomicsTK' && ( startsWith(github.ref, 'refs/tags/v') || github.ref == 'refs/heads/master' )
+    strategy:
+      matrix:
+        python: ["3.7", "3.8", "3.9", "3.10", "3.11"]
+    steps:
+      - uses: actions/checkout@v3
+      - uses: actions/setup-python@v4
+        with:
+          python-version: ${{ matrix.python }}
+      - name: Install tox
+        run: |
+          pip install --upgrade pip
+          pip install tox tox-gh-actions
+      - name: Run tox
+        env:
+          PYTEST_ADDOPTS: "--durations=0 -vv"
+        run: |
+          tox
+  lint:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+      - uses: actions/setup-python@v4
+        with:
+          python-version: "3.11"
+      - uses: nikeee/setup-pandoc@v1
+      - name: Install tox
+        run: |
+          pip install --upgrade pip
+          pip install tox
+      - name: Run tox for lint and docs
+        run: |
+          tox -e flake8,docs
+      - uses: actions/upload-artifact@v3
+        with:
+          name: docs
+          path: ./docs/_build
+  build_wheels:
+    name: Build wheels on ${{ matrix.buildplat[0] }} / ${{ matrix.buildplat[1] }} for ${{ matrix.python }}
+    runs-on: ${{ matrix.buildplat[0] }}
+    # Don't build external PRs.
+    if: github.repository == 'DigitalSlideArchive/HistomicsTK'
+    strategy:
+      matrix:
+        buildplat:
+          - [ubuntu-latest, manylinux_x86_64]
+          - [ubuntu-latest, musllinux_x86_64]
+          - [macos-latest, macosx_*]
+          - [windows-latest, win_amd64]
+        python: ["cp37", "cp38", "cp39", "cp310", "cp311"]
+        exclude:
+          - buildplat:
+            - [windows-latest, win32]
+      fail-fast: false
+
+    steps:
+      - uses: actions/checkout@v3
+        with:
+          fetch-depth: 0
+
+      - run: git status && git tag --list && pwd
+      - uses: actions/setup-python@v4
+        with:
+          python-version: "3.x"
+
+      - name: Build wheels
+        uses: pypa/cibuildwheel@v2.14.1
+        env:
+          CIBW_BUILD: ${{ matrix.python }}-${{ matrix.buildplat[1] }}
+
+      - uses: actions/upload-artifact@v3
+        with:
+          name: ${{ matrix.python }}-${{ startsWith(matrix.buildplat[1], 'macosx') && 'macosx' || matrix.buildplat[1] }}
+          path: ./wheelhouse/*.whl
+
+  # Still on Circle-CI
+  #   - docker
+  #   - publish-docker
+  #   - docs-deploy
+
+  release:
+    runs-on: ubuntu-latest
+    needs:
+      - test
+      - lint
+      - build_wheels
+    if: github.repository == 'DigitalSlideArchive/HistomicsTK' && ( startsWith(github.ref, 'refs/tags/v') || github.ref == 'refs/heads/master' )
+    environment:
+      name: pypi
+      url: https://pypi.org/p/histomicstk
+    permissions:
+      id-token: write
+    steps:
+      - uses: actions/download-artifact@v3
+        with:
+          name: artifact
+          path: dist
+      - uses: pypa/gh-action-pypi-publish@release/v1

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -3,7 +3,6 @@ cmake_minimum_required(VERSION 3.6.2)
 project(histomicstk)
 
 find_package(PythonInterp REQUIRED)
-find_package(PythonLibs REQUIRED)
 find_package(PythonExtensions REQUIRED)
 
 add_subdirectory(histomicstk/features)

--- a/setup.py
+++ b/setup.py
@@ -30,7 +30,7 @@ def prerelease_local_scheme(version):
     """
     from setuptools_scm.version import get_local_node_and_date
 
-    if os.getenv('CIRCLE_BRANCH') in {'master'}:
+    if os.getenv('CIRCLE_BRANCH') in {'master'} or os.getenv('CI_BRANCH') in {'master'}:
         return ''
     else:
         return get_local_node_and_date(version)

--- a/tox.ini
+++ b/tox.ini
@@ -10,6 +10,7 @@ toxworkdir = {toxinidir}/.tox
 passenv =
   DOCKER_*
   GENERATE_GROUNDTRUTH
+  PYTEST_*
 # This adds the tests directory to the python path so we can import the test
 # utilities as needed.
 setenv =

